### ISSUE TITLE
Scheduling early transition to Workbench

### ISSUE THREAD

**Issue by tobyhodges** [swcarpentry/python-novice-inflammation#1032]
Short version: **are you are willing to volunteer your lesson for early transition to The Carpentries Workbench 
infrastructure?**

---

Longer version below:

As I hope you are already aware, @zkamvar and the rest of the Curriculum Team are preparing to roll out the new lesson infrastructure, [The Carpentries Workbench](https://carpentries.github.io/workbench/), across all of The Carpentries official lessons in early May 2023. This means that all Data Carpentry, Library Carpentry, and Software Carpentry lesson repositories will be modified to adopt the new infrastructure at the end of this month.

As you might imagine, coordinating a rollout like this involves a large amount of time and effort, and one thing that will really help us to keep to the schedule and avoid disruption for the community is to be able to **prepare some lessons for 
transition before the 1st May deadline**.

With this in mind, **I invite you to volunteer to schedule an early transition for theis lesson repository.** If you expect to have time to adopt the Workbench version of the lesson repository in the next couple of weeks, please reply to let me know. As lesson Maintainers, these are the things you will need to be prepared to do for the transition:
1. **Be prepared to quickly merge pull requests from Zhian**. These pull requests will help update the lesson and aid the transition process.
2. **Close any outstanding pull requests.** Your repository currently has very few or zero open PRs, but we cannot rule out the possibility that someone will open more in the time between now and when the transition takes place. [Open PRs will be invalidated when the transition takes place](https://carpentries.github.io/sandpaper-docs/pull-request.html#transition-from-carpentriesstyles).
3. **Preview the lesson (in the https://github.com/fishtree-attempt organisation) when it is available**. Zhian will open an issue in your repository, tagging you when the preview is available. You should reply on that issue if you notice any problems with the Workbench version of the lesson site and repository in the preview.
4. Just before the transition happens, the repository will be temporarily set as read-only and an issue will be opened (see an example from the Maintainer Onboarding curriculum: https://github.com/carpentries/maintainer-onboarding/issues/69) 
5. After the transition, **delete and recreate any forks and local clones** you have of the repository and then **confirm that you have done so** (you will be given specific instructions in an issue).

If you are willing and able to help us ensure a smooth rollout of the Workbench, please reply to let me know that you are happy to volunteer your lesson for early transition. If there is a specific date you would like the transition to take place (or that we should avoid), please also mention that.

**Comment by zkamvar** [comment]
Thank you for participating in the early transition to The Workbench. I have been able to render a preview of your lesson at https://fishtree-attempt.github.io/python-novice-inflammation.

I am tracking issues for your lesson in https://github.com/carpentries/lesson-transition/issues/22. I will be updating the preview next week as I work on any remaining issues. 

Please inspect the preview and make note in https://github.com/carpentries/lesson-transition/issues/22 if you see anything that needs to be fixed (e.g extra templating, narrow tables, or malformed text).

**Comment by tobyhodges** [comment]
Hi @noatgnu @ineelhere we are getting ready for an early transition for this lesson, and we have prepared [a schedule for early lesson transitions](https://docs.google.com/spreadsheets/d/1Y_ZdAZu11Lc1bcInE9801ax6khvgFzjTh70P3v8crL8/edit?usp=sharing). 

Please check that document to see the date we have planned for transitioning this lesson, and adjust the sheet if a different date would be more convenient for you (no more than three lessons per day, please!). The README sheet gives more information. If we don't hear from you, we will go ahead with the transition on the date shown in that document.

**Comment by zkamvar** [comment]
Hi Both, I just found an issue that my transition process did not catch:

In the workbench, all supporting files and folders live underneath episodes: https://carpentries.github.io/workbench/transition-guide.html#folder-structure. This was an effort to simplify the various paths for the lesson. 

However, in this repository, you have the top-level code folder, which is retained in the workbench, but it does not get propogated to the website. 

I was planning to transition this lesson today, but I will have to make an executive decision to move the `code/` folder into `episodes/files/code/` and update the links in the setup and instructor notes. 

I apologise for not flagging this earlier.

**Comment by zkamvar** [comment]
I apologise, I was unable to make the transition happen today. There is nothing impeding the transition itself, it's just a matter of a lot of outside factors coming together in ways that took away my time. The transition process itself takes ~ 5 minutes total, but as you can imagine, it's a very stressful transition on my part because I am changing something that a lot of people have put a lot of time an effort into. Thus, I like to give myself at least two hours for the transition to happen so that I can correct anything if it goes wrong.

I will try again tomorrow. Thank you for your patience.

**Comment by zkamvar** [comment]
This lesson will be converted to use [The Carpentries Workbench][workbench]
To prevent accidental reversion of the changes, we are temporarily revoking
write access for all collaborators on this lesson:

 - [ ] @richmccue (push)
 - [ ] @coatless (push)
 - [ ] @annajiat (push)
 - [ ] @noatgnu (push)
 - [ ] @vinisalazar (push)
 - [ ] @mxiang1 (push)
 - [ ] @ineelhere (push)
 - [ ] @Gina123-P (push)

If you no longer wish to have write access to this repository, you do not
need to do anything further.

1. What you can expect from the transition 📹: https://carpentries.github.io/workbench/beta-phase.html#beta
2. How to update your local clone 💻: https://carpentries.github.io/workbench/beta-phase.html#updating-clone
3. How to update (delete) your fork (if you have one) 📹: https://carpentries.github.io/workbench/faq.html#update-fork-from-styles

If you wish to regain write access, please re-clone the repository on your machine and
then comment here with `I am ready for write access :rocket:` and the
admin maintainer of this repository will restore your permissions.

If you have any questions, please reply here and tag @zkamvar

[workbench]: https://carpentries.github.io/workbench

**Comment by zkamvar** [comment]
The deed is done. The infrastructure takes a few minutes to bootstrap and cache the packages for the lesson build. Once the build is done, I will switch github pages to deploy from the `gh-pages` branch and you will have your workbench lesson. 

Thank you all for your enthusiasm and your patience!

**Comment by zkamvar** [comment]
The Workbench version is now live: https://swcarpentry.github.io/python-novice-inflammation/

In addition, here is [map of commits that were changed during the transition](https://github.com/carpentries/lesson-transition/blob/release_swc/python-novice-inflammation/release/swcarpentry/python-novice-inflammation-commit-map.hash)

**Comment by zkamvar** [comment]
Access granted, @ineelhere!

**Comment by ErinBecker** [comment]
Hi @noatgnu - You have not yet responded here to report that you completed the steps listed by @zkamvar above to regain your Maintainer access to this repository. The lesson infrastructure transition required a second Maintainer team to be temporarily created for all lessons, which can only be removed when all Maintainers have confirmed that they are ready to have their access restored. Maintenance of both teams is not sustainable for The Carpentries in the long run. At the beginning of **November 2023** (note the deadline has been extended) we will remove access for Maintainers who have not taken the necessary action to continue working on the repository.

The Curriculum Team is truly grateful for everything you have already done as a Maintainer and we do not want to lose you! But we also understand that circumstances change and you may no longer have capacity to continue in your role. If you wish to retain your access and keep supporting the community, please follow the steps above then confirm that you are ready by commenting I am ready for write access 🚀 on this thread before the end of your day on **Wednesday 1st November.**

**Comment by noatgnu** [comment]
Hi,

I am ready for write access. I thought I would be granted before since I was there and the first one to recommend the early transition.

Toan

**Comment by vinisalazar** [comment]
Hi,

I'm ready for write access :)

Vini

**Comment by zkamvar** [comment]
@noatgnu your access has been restored. I apologise for the miscommunication regarding access; it was revoked for all maintainers because of the risk of accidentally merging and pushing changes from local clones that were not yet reset.

**Comment by tobyhodges** [comment]
Thanks @vinisalazar. As your access was associated with your previous activity as Maintainer Community Lead (thank you!) and you are not a Maintainer for this lesson, The Curriculum Team will take this opportunity to remove your elevated permissions again. This helps us restore consistency and improve security of the repository, ensuring that only current Maintainers have elevated privileges on the project.

**Comment by vinisalazar** [comment]
All good, thank you for clarifying @tobyhodges :)
